src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/BotSegmentsPlaybackController.cs

```diff
@@ -108,6 +110,21 @@ private void ProcessBotSegments()
                         ProcessBotSegmentAction(nextBotSegment, transformStatuses, entityStatuses);
                     }
 
+                    // Run (and potentially end) the validations for this segment
+                    nextBotSegment.ProcessValidation();
+
+                    // Also run the validations for the botsegmentList if they exist
+                    foreach (var validation in _botSegmentListValidations)
+                    {
+                        validation.ProcessValidation(nextBotSegment.Replay_SegmentNumber);
```

Comment:
So this is passing down the segment number, but the validation is not really part of that segment, it is part of that segment list as a whole... do we think this could confuse the user? This is why I used -1 for sequences, and then in the logs that gets converted to "SEQUENCE"

Reply:
i want to see the active segment in the logs as this runs so I can see what segment I was on as the validation was running

Reply:
they are absolutely being evaluated during that currently active segment, and should log as such

Reply:
Sounds good to me 👍 I can get behind that logic